### PR TITLE
Support (non-)transitional virtio-balloon 

### PR DESCRIPTION
**Pull request by andreabolognani**
**What this PR does / why we need it**:

Follow-up to https://github.com/kubevirt/kubevirt/pull/4730

There was a bug in libvirt that prevented virtio-balloon to be treated like other virtio devices, but that's been addressed now.

**Special notes for your reviewer**:

The mirror URLs (`https://storage.googleapis.com/builddeps/...`) for the new RPMs are not generated, because I have no idea how to make that happen and I probably don't even have the necessary permissions anyway O:-)

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
> The mirror URLs (`https://storage.googleapis.com/builddeps/...`) for the new RPMs are not generated, because I have no idea how to make that happen and I probably don't even have the necessary permissions anyway O:

The mirror urls will be generated in a separate PR by our bot once this is merged, no action required.

File: pkg/virt-launcher/virtwrap/converter/converter_test.go [hunk 1]
```diff
@@ -622,15 +622,15 @@ var _ = Describe("Converter", func() {
   <iothreads>3</iothreads>
 </domain>`, domainType, "%s")
 		var convertedDomainWith5Period = fmt.Sprintf(convertedDomain,
-			`<memballoon model="virtio">
+			`<memballoon model="virtio-non-transitional">
```
Comment (rmohr) [inline pkg/virt-launcher/virtwrap/converter/converter_test.go:625]: Threr exists a unit test her which uses a function called `ExpectVirtioTransitionalOnly`, would you mind adjusting the checks in this function to also check the ballooning device?

**Review by rmohr — CHANGES_REQUESTED** [review]
Looks greast. One small request.

File: pkg/virt-launcher/virtwrap/converter/converter_test.go (hunk 1 above)
Comment (rmohr) [reply pkg/virt-launcher/virtwrap/converter/converter_test.go:625]: Never mind, you did that.

**Review by rmohr — APPROVED** [review]
My mistake, the test in question is already adjusted.

/lgtm
/approve

**Comment by rmohr** [comment]
/lgtm cancel

The e2e result show 3 tests which need some more minor fixes.

**Comment by andreabolognani** [comment]
@rmohr Fixed the functests, thanks for pointing the failure out!

**Comment by andreabolognani** [comment]
@rmohr unfortunately a cherry-pick is not going to work, because 0.36 doesn't have bazeldnf :'(

One possible solution would be to make an ad-hoc rebuild of the kubevirt/libvirt container so that it includes the updated packages... While the kubevirt/libvirt repository is currently deprecated and Travis CI has become nigh useless in the past few months, everything is still set up, so we could create a branch there for this purpose.

What do you think?

**Comment by rmohr** [comment]
> One possible solution would be to make an ad-hoc rebuild of the kubevirt/libvirt container so that it includes the updated packages... While the kubevirt/libvirt repository is currently deprecated and Travis CI has become nigh useless in the past few months, everything is still set up, so we could create a branch there for this purpose.
> 
> What do you think?

Please wait a little with that. I just wanted to have the PR prepared, but I don't think that we can merge it. Everything which is merged on that branch causes rebuilds in osbs. As far as I understand we don't have 8.3.1 yet available, so we would basically deliver a version which we can't yet use.

**Comment by andreabolognani** [comment]
Alright, I'll let you handle it... Just let me know if I can be of assistance :)

**Comment by rmohr** [comment]
> Alright, I'll let you handle it... Just let me know if I can be of assistance :)

Thanks. At some point we will need the rebuild for 0.36.

**Pull request merged**